cfe_internal/update/update_policy.cf

```diff
@@ -597,4 +610,64 @@ body delete u_tidy
       rmdirs   => "true";
 }
 
+bundle agent install_modules
+{
+  vars:
+    "vendored_dir" string => "$(sys.inputdir)$(const.dirsep)modules$(const.dirsep)packages$(const.dirsep)vendored$(const.dirsep)";
+    "override_dir" string => "$(sys.inputdir)$(const.dirsep)modules$(const.dirsep)packages$(const.dirsep)";
+    "vendored_paths" slist => findfiles("$(vendored_dir)*.mustache");
+    "package_paths_tmp" slist => findfiles("$(override_dir)*");
+    "package_paths" slist => filter("$(override_dir)vendored", package_paths_tmp, "false", "true", 999);
+    "vendored_modules" slist => maplist(regex_replace("$(this)", "$(vendored_dir)(.*).mustache", "$1", "g"), @(vendored_paths));
+    "override_modules" slist => maplist(regex_replace("$(this)", "$(override_dir)(.*)", "$1", "g"), @(package_paths));
+
+  classes:
+    "override_vendored_module_$(override_modules)" expression => fileexists("$(override_dir)$(override_modules)");
+    "module_$(override_modules)" expression => fileexists("$(override_dir)$(override_modules)");
+
+# NOTE: here we only send `bindir` into the template
+# this helps since we also rely on makefiles to replace
+# the python shebang for tests, so need to limit "what we change"
+# in these packages modules, instead of providing "everything"
+# we could send to the template via `datastate()`
+
+  files:
+    "$(sys.workdir)/modules/packages/$(vendored_modules)"
+      create => "true",
+      perms => u_mo("755", "root"),
+      unless => "canonify(override_vendored_module_$(vendored_modules))",
+      edit_template => "$(vendored_dir)$(vendored_modules).mustache",
+      template_method => "mustache",
+      template_data => parsejson('{"bindir":"$(sys.bindir)"}');
+
+    "$(sys.workdir)/modules/packages/$(override_modules)"
+      copy_from => u_cp_missing_ok("$(override_dir)$(override_modules)"),
+      perms => u_mo("755", "root"),
+      if => or (
+        canonify("override_vendored_module_$(override_modules)"),
+        canonify("module_$(override_modules)"));
+
+    "$(sys.workdir)/modules"
+      comment => "Copy any non-packages modules",
+      handle => "cfe_internal_update_policy_files_nonpackages_modules",
+      copy_from => u_cp("$(sys.inputdir)$(const.dirsep)modules"),
+      if => fileexists("$(sys.inputdir)$(const.dirsep)modules"),
+      depth_search => u_recurse("inf"),
+      perms => u_mo("755", "root"),
+      action => u_immediate,
+      file_select => not_vendored_modules;
+
+
+  reports:
+    "vendored_dir: $(vendored_dir)";
```

Comment:
Should strip these or at least DEBUG class guard them.

Reply:
yes, a todo for me still :+1: